src/tower/game_view.py

```diff
@@ -50,3 +62,82 @@ def on_draw(self):
 
         self.towers.draw()
         self.towers[0].show_range()
+
+        self.enemies.draw()
+
+    def on_update(self, delta_time: float):
+        """Updates the position of all enemies"""
+        self.enemies.update()
+
+    def calculate_enemy_path(self) -> list[tuple[int]]:
+        """calculates the path enemies should follow
+        returns a list of tuples (x, y) representing a coordinate of the path
+        that the enemy follows
+        """
+        # contains all coordinates in the form (x, y), that the enemies should follow
+        position_list = []
+
+        # this is a 1D representation of our map containing sprites
+        all_sprites = self.scene.get_sprite_list("Tile Layer 1")
+
+        # this is a 2D representation of our map containing only the id of the sprite
+        tile_map = self.tile_map.get_tilemap_layer("Tile Layer 1").data
+
+        # get the postion of the start block
+        row, col = self._find_start_index(tile_map)
+
+        # visited keeps track whether we already visited tile_map[row][col]
+        visited = [[False for _ in range(len(tile_map[0]))] for _ in range(len(tile_map))]
+
+        # while the end of the path is not reached
+        while all_sprites[row * len(tile_map[0]) + col].properties["tile_id"] != 5:
+            # add current position to position list
+            # get sprite of row and col by calculating its 1D position
+            sprite = all_sprites[row * len(tile_map[0]) + col]
+
+            position_list.append((sprite.center_x, sprite.center_y))
+            visited[row][col] = True
+
+            # get next path position
+            row, col = self.get_next_path_position(row, col, tile_map, visited)
+
+        sprite = all_sprites[row * len(tile_map[0]) + col]
+        position_list.append((sprite.center_x, sprite.center_y))
+        return position_list
+
+    def _find_start_index(self, tile_map):
+        """returns the index of the tile that represents the start of the path"""
+        start_sprites = [5]
+        for row_index, row in enumerate(tile_map):
+            for col_index, col in enumerate(row):
+                if col in start_sprites:
+                    return row_index, col_index
+        return None, None
+
+    def get_next_path_position(self, row, col, tile_map, visited) -> int:
+        """Returns the next path position
+        assumes that there is only one possible path
+        """
+        # has all id's of paths
+        path_blocks = [3, 6, 5]
```

Comment:
should we move this to constants to keep track which of our blocks are paths?

Reply:
yeah that's a good idea

Reply:
make it a set as well